File: Longest_Increasing_Subsequence.c
int lengthOfLIS(int* nums, int numsSize) {
    if (numsSize == 0) return 0;

    int dp[numsSize];
    int maxLength = 1;

    for (int i = 0; i < numsSize; i++) {
        dp[i] = 1;
        for (int j = 0; j < i; j++) {
            if (nums[i] > nums[j]) {
                dp[i] = dp[i] > dp[j] + 1 ? dp[i] : dp[j] + 1;
            }
        }
        maxLength = maxLength > dp[i] ? maxLength : dp[i];
    }
    return maxLength;
}

File: Longest_Increasing_Subsequence.py
class Solution(object):
    def lengthOfLIS(self, nums):
        temp = [1] * len(nums)
        max_element = 1
        
        for i in range(len(nums)):
            temp[i] = 1
            for j in range(i):
                if nums[j] < nums[i]:
                    temp[i] = max(temp[i],temp[j] + 1)
            max_element = max(temp[i],max_element)
        return max_element